### PR TITLE
Fix exception documentation on System.Collections.Immutable

### PR DESCRIPTION
**Pull request by AArnott**
I noticed in making another change that a bunch of incorrect xml docs describing exceptions existed. This fixes those up by removing the invalid ones and adding explanations for the valid ones.

### CODE REVIEW

File: src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList`1.cs [hunk 1]
```diff
@@ -1145,7 +1146,7 @@ System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
         /// </summary>
         /// <param name="index">The index.</param>
         /// <param name="item">The item.</param>
-        /// <exception cref="System.NotSupportedException"></exception>
+        /// <exception cref="NotSupportedException">Always thrown from the setter.</exception>
```
Comment (stephentoub) [inline src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList`1.cs]: "setter" doesn't seem like the right description here

File: src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList`1.cs [hunk 2]
```diff
@@ -1155,7 +1156,7 @@ void IList<T>.Insert(int index, T item)
         /// Removes the value at the specified index.
         /// </summary>
         /// <param name="index">The index.</param>
-        /// <exception cref="System.NotSupportedException"></exception>
+        /// <exception cref="NotSupportedException">Always thrown from the setter.</exception>
```
Comment (stephentoub) [inline src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList`1.cs]: Ditto

**Comment by stephentoub** [comment]
Two minor nits, otherwise LGTM.

**Comment by AArnott** [comment]
Thanks for the review. I'll merge after automated checks complete.

**Pull request merged**